### ISSUE TITLE
create a communication class for handling signaling and the communication between threads

### ISSUE THREAD

**Issue by Marzona** [Marzona/rig-remote#34]
We need to define how the threads will exchange commands and status, as we agreed we need to create a class that provides 2 queues (of size 1 for now) and the methods for handling the insertion/deletion of events. The class should provide the signaling logic and provide methods for creating/sending messages  too.

This is going to be a dedicated module

**Comment by MaineTim** [comment]
I've coded out two classes as a skeleton module, a lower-level class that instantiates the two queues and provides access methods for both numeric signals and generic objects, and a higher-level scan-thread comm class that applies the lower-level class to gui-event updates and scan-thread status notification. I've substituted it into the program code and just need to add some processing code in the main thread to handle the button reset (just looking at how to do that most efficiently). Right now limited passes terminates as it should,  but of course the button doesn't update. No lock-ups, however, so that's progress. As soon as I've added the last piece and have the button resetting,  I'll upload to a sub-branch and we can see if it'll be a basis for more fleshing out, or if you want to go in another direction. But now dinner...
